src/parser.rs

```diff
@@ -18,320 +18,52 @@
-                        } else {
-                            Level::Info
-                        },
-                        tag: "googletest".to_owned(),
-                        process,
-                        message: line[12..].trim().to_owned(),
-                        ..Default::default()
-                    })
-                } else {
-                    Err(err_msg("Failed to parse gtest: Missing closing bracket"))
-                }
-            } else {
-                Err(err_msg("Failed to parse gtest: Missing opening bracket"))
-            }
-        } else {
-            Err(err_msg("Failed to parse gtest: Message is too short"))
-        }
-    }
-
-    fn parse_bugreport(line: &str) -> Result<Record, Error> {
-        if line.starts_with('=')
-            || line.starts_with('-')
-            || (line.starts_with('[') && line.ends_with(']'))
-        {
-            if line.chars().all(|c| c == '=') {
-                Ok(Record {
-                    level: Level::Info,
-                    message: line.to_owned(),
-                    raw: line.to_owned(),
-                    ..Default::default()
-                })
-            } else if line.starts_with("== ") {
-                Ok(Record {
-                    level: Level::Info,
-                    message: line[3..].to_owned(),
-                    raw: line.to_owned(),
-                    tag: line[3..].to_owned(),
-                    ..Default::default()
-                })
-            } else if line.is_empty() {
-                Err(err_msg("Unparseable"))
-            } else if let Ok((_, (tag, value))) = property(CompleteStr(line)) {
-                Ok(Record {
-                    message: value,
-                    tag,
-                    raw: line.to_owned(),
-                    ..Default::default()
-                })
-            } else {
-                let line = line.trim_matches('=').trim_matches('-').trim();
-                match bugreport_section(CompleteStr(line)) {
-                    Ok((_, (tag, message))) => Ok(Record {
-                        level: Level::Info,
-                        message,
-                        raw: line.to_owned(),
-                        tag,
-                        ..Default::default()
-                    }),
-                    Err(e) => Err(format_err!("{}", e)),
-                }
-            }
-        } else {
-            Err(err_msg("Unparseable"))
+    pub fn with_default_rules() -> Self {
+        Parser {
+            parsers: vec![
+                Box::new(formats::DefaultParser {}),
+                Box::new(formats::MindroidParser {}),
+                Box::new(formats::CsvParser {}),
+                Box::new(formats::JsonParser {}),
+                Box::new(formats::GTestParser {}),
+                Box::new(formats::BugReportParser {}),
+            ],
+            last: None,
         }
     }
 
     pub fn parse(&mut self, line: String) -> Record {
```

Comment:
Think changing this to `&str` can avoid to move the `String`. There is no need to actually move the line.